python/ray/serve/_private/deployment_scheduler.py

```diff
@@ -164,6 +181,12 @@ def is_non_strict_pack_pg(self) -> bool:
         )
 
 
+@dataclass
+class LaunchingReplicaInfo:
+    az: Optional[str] = None
+    node_id: Optional[str] = None
```

Comment:
AZ is not used here, and `node_id` is only used for compact scheduling (vs normal spread scheduling). If both are None then it just means we're not using `launching_replicas` for any special scheduling. (launching replicas is also best effort, it contains the information that we "plan" on scheduling a new replica to before we get the round trip update from the GCS, but it's not guaranteed)

Reply:
az being optional makes sense

but for node_id, shouldn't it always be populated but just ignored by spread scheduling? or else this entire data structure unused?

Reply:
Hmm, for spread scheduling the Serve controller doesn't look at the current cluster view / available nodes at all and instead lets core select a node that satisfies requirements (like resources, PG, etc), so we can't really populate it with anything (and also the calculations in `get_available_resources_per_node` will be inaccurate if we just randomly populate it). Whereas with compact scheduling we are selecting a node ourselves which is why we populate it.

I think the options here are set it to `None` or set it to `LaunchingReplicaInfo(az=None, node_id=None)`. Would you prefer the former?

Reply:
I see now, because this is populated in the shared method that both codepaths go through, but for spread case we won't actually target a node. In this I'm OK with it as-is, but I would suggest documenting that these are both best-effort and perhaps renaming the fields to something like `target_node_id` and `target_az`.

Reply:
Makes sense, I've added missing docstrings and renamed them to `target_node_id` and `target_az`!